src/scripts/main.js

```diff
@@ -1 +1,74 @@
 'use strict';
+
+const firstPromise = new Promise((resolve, reject) => {
+  let resolved = false;
+
+  const timeout = setTimeout(() => {
+    if (!resolved) {
+      reject(new Error('First promise was rejected'));
```

Comment:
It's not a good practice to reject a promise with a string. Instead, reject it with an instance of Error. This provides a stack trace and helps to debug the code. Change this line to `reject(new Error('First promise was rejected'));`